monkey/monkey_island/cc/models/monkey.py

```diff
@@ -84,13 +88,52 @@ def get_os(self):
             os = "windows"
         return os
 
+    @staticmethod
+    @ring.lru()
+    def get_label_by_id(object_id):
```

Comment:
Why doesn't this need a cache deletion flow? 
The label can change if the monkey hostname changes.
in that case, the hostname cache would be cleared but not the label cache.

Basic solution would be adding the cache clear line to set_hostname